examples/all-devices-app/esp32/main/DeviceShellCommands.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <DeviceShellCommands.h>
+#include <lib/shell/streamer.h>
+
+void InitServerWithDeviceType(std::string deviceType);
+
+namespace chip {
+namespace Shell {
+
+Shell::Engine DeviceCommands::sSubShell;
+
+void DeviceCommands::Register()
+{
+    static const shell_command_t subCommands[] = {
+        { &SetDeviceTypeHandler, "set", "Usage: device set <device-type>" },
+    };
+    sSubShell.RegisterCommands(subCommands, MATTER_ARRAY_SIZE(subCommands));
+
+    // Register the root `device` command in the top-level shell.
+    static const shell_command_t deviceCommand = { &DeviceHandler, "device", "Device type management commands" };
```

Comment:
Does this not conflict with this "device" command? https://github.com/project-chip/connectedhomeip/blob/99d86808c347dba0c99be7d533459f9153766333/src/lib/shell/commands/Device.cpp#L40